src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -135,7 +114,25 @@ public class PipelineCacheServer
 
                 if (!string.IsNullOrEmpty(cacheHitVariable))
                 {
-                    context.SetVariable(cacheHitVariable, result != null ? "true" : "false");
+                    if (result == null) {
+                        context.SetVariable(cacheHitVariable, "false");
+                    }  else  {
+                        context.Verbose($"Exact fingerprint: `{result.Fingerprint.ToString()}`");
+
+                        bool foundExact = false;
+                        foreach(var fingerprint in fingerprints)
+                        {
+                            context.Verbose($"This fingerprint: `{fingerprint.ToString()}`");
+
+                            if(fingerprint == result.Fingerprint)
+                            {
+                                foundExact = true;
+                                break;
+                            }
+                        }
+
+                        context.SetVariable(cacheHitVariable, foundExact ? "true" : "inexact");
```

Comment:
>inexact [](start = 85, length = 7)

Wording?
Guidance should be to either eq/ne on true/false (i.e. don't eq/ne on inexact)